core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java

```diff
@@ -511,17 +511,19 @@ private QueryAssert hasPlan(PlanMatchPattern expectedPlan, Consumer<Plan> additi
 
             if (!skipResultsCorrectnessCheckForPushdown) {
                 // Compare the results with pushdown disabled, so that explicit matches() call is not needed
-                verifyResultsWithPushdownDisabled();
+                hasCorrectResultsRegardlessOfPushdown();
```

Comment:
Since `hasCorrectResultsRegardlessOfPushdown()` returns a QueryAssert, shouldn't this be:

```
return hasCorrectResultsRegardlessOfPushdown()
```

?

Reply:
`QueryAssert` needs to know whether `hasCorrectResultsRegardlessOfPushdown` returns `this` for chaining (convenience) or a different `QueryAssert` state. Without knowing it, it's impossible to judge whether returning `this` is "more correct" than returning `hasCorrectResultsRegardlessOfPushdown()`

Reply:
Doesn't it know? It's exposed as a public method to be used by tests, so the contract should be clear.

Regardless, I find it a bit confusing. When I read the code it's not immediately clear whether there's a bug lurking (is the result of `hasCorrectResultsRegardlessOfPushdown()` eager or lazy, so is it possible it's a no-op?)

Reply:
It would be more confusing to conditionally `return hasCorrectResultsRegardlessOfPushdown()` else `return this`.

Reply:
How so? If it doesn't go into that block, there's nothing else to return, and no potential additional state induced by that method, so `this` is the only natural thing to return.

Reply:
> so `this` is the only natural thing to return.

yes!
and the code needs to know (and does know) that `hasCorrectResultsRegardlessOfPushdown` returns `this`, so leverages this fact.